### PR TITLE
BSP-194: UpdateItemInstance handler

### PR DESCRIPTION
**Pull request by 0x103**
- UpdateItemInstance handler  + DI
- OpenAPI doc updated.
- KnexItemInstanceRepository updated for UpdateItemInstance function.
- Custom error type added in common for SQL update errors.
- Instance update event created.
- Custom update error added in assignItem & updateItemInstance functions.
- New Tezos event added.
- New Tezos service function for instance update.
- Tezos/TezosTokenizationService DI updated to add ItemInstanceRepository.

### CODE REVIEW

File: packages/item-service/src/openapi-docs.json [hunk 1]
```diff
@@ -231,6 +231,56 @@
...
+                                        "example": 1
+                                    },
+                                    "instance_number": {
+                                        "type": "integer",
+                                        "example": 1
+                                    },
+                                    "data": {
+                                        "type": "object",
+                                        "example": {
+                                            "level": 200,
```
Comment (podefr) [inline packages/item-service/src/openapi-docs.json:258]: level should be a string "200" otherwise we get an error (data fields can only be strings)

Comment (0x103) [reply packages/item-service/src/openapi-docs.json:258]: Indeed yes! Thank you.

File: packages/item-service/src/events/item.ts [hunk 2]
```diff
@@ -30,3 +30,14 @@ export class ItemAssignedEvent implements EventBusEvent {
         this.user_id = user_id;
     }
 }
+
+export class ItemInstanceUpdatedEvent implements EventBusEvent {
+    name = "ItemInstanceUpdatedEvent";
+    readonly item_id: number;
+    readonly instance_number: number;
+
+    constructor(item_id: number, instance_number: number) {
```
Comment (tkjwa) [inline packages/item-service/src/events/item.ts:39]: You could add a "new_data" field here and in the tokenisation listener you wouldn't need to fetch the item from the repo

Also to ease writing DTO's, you can do:
```
export class ItemInstanceUpdatedEvent implement EventBusEvent {
    name = "ItemInstanceUpdatedEvent"
    constructor(readony item_id: number, readyonly instance_number) {}
}
```

File: packages/js-common/src/errors/SQLUpdateNoRowsAffected.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class SQLUpdateNoRowsAffected extends Error {
```
Comment (tkjwa) [inline packages/js-common/src/errors/SQLUpdateNoRowsAffected.ts:1]: Why not export default ?

File: packages/js-common/src/errors/index.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+export { SQLUpdateNoRowsAffected } from "./SQLUpdateNoRowsAffected";
```
Comment (tkjwa) [inline packages/js-common/src/errors/index.ts:1]: Can you explain me the benefits of this index please ?

**Review by tkjwa — APPROVED** [review]

File: packages/item-service/src/events/item.ts (hunk 2 above)
Comment (0x103) [reply packages/item-service/src/events/item.ts:39]: Ahah, I hesitated to do that and in the end I choose to do like we did everywhere else, but I'll update that, it'll save a query !

**Pull request merged**